Fix RBAC for Folders (descendent of VMs)

kbrock opened the pull request:
Part of https://github.com/ManageIQ/manageiq/issues/21516

We preload to avoid N+1 in our map
But some of the relations are not associations
(i.e.: parent_blue_folders and resource_pool)

So we are only using preload for those that are associations.

This fixes:
```
DEPRECATION WARNING: Vm.parent_blue_folders does not exist (called from preload at manageiq/lib/miq_preloader.rb:7)
```

And fixes a rails 6.1 blowup

jrafanie commented on lib/rbac/filterer.rb:
```diff
@@ -717,7 +717,10 @@ def matches_via_descendants(klass, descendant_klass, options)
+          r_klass = descendants.try(:klass) || descendants.first.class
+          if r_klass.reflections.keys.include?(method_name.to_s)
+            MiqPreloader.preload(descendants, method_name)
+          end
```
Does it make sense to link this to the `parent_blue_folders` and vice versa?  Is this only here for that one fake association?

jrafanie replied:
In other words, "this stuff is here for one specific reason, here's that one place that needs it..."

kbrock replied:
this is here for 2 out of the 3 implementions. `ResourcePool#resource_pool`, and `Vm#_blue_folder`

I did put that in the commit

kbrock commented:
push:
- fixed rubocops
- added comments per LJ to make reasoning more obvious

push:
- fixed rubocops. they sure are picky about the aliases. I can throw those away. they are only used in tests but they read better for me.
